app/models/course_data/block.rb

```diff
@@ -23,6 +25,8 @@ class Block < ApplicationRecord
   belongs_to :week
   has_one :course, through: :week
   serialize :training_module_ids, Array
+  before_update :editable?, if: :content_fields_changed?
+  before_destroy :deletable?
```

Comment:
These blocks should still be deleted if an instructor is deleting the entire timeline (so that they can go throught the wizard again), but they shouldn't be able to delete the week containing an indestructible block, nor the block directly.

This will probably shake out in the feature specs, once we configure the wizard with the relevant blocks.

Reply:
Sorry, I'm not sure I quite understand this first part. The request is:

1. If an instructor is deleting all of the blocks, they should be able to delete the block even if `deletable?` returns false
1. If an instructor is attempting to delete a week that contains an indestructible block, the entire week should not be able to be deleted. (?)
1. The instructor should not be able to delete an indestructible block.

Did I get all the use cases right?

Reply:
One option for this is just to ignore validation on the server side. It seems odd someone with access to edit the backend would try and go around the UI.

Reply:
Yeah, you got all those use cases right. There's a feature to delete the entire timeline, which we still need to keep as an option. But deleting a week with an indestructible block should be impossible for an instructor on the frontend.